src/PowerShellEditorServices/Services/PowerShell/Execution/SynchronousPowerShellTask.cs

```diff
@@ -65,8 +61,14 @@ public override IReadOnlyList<TResult> Run(CancellationToken cancellationToken)
                     _psesHost.WriteWithPrompt(_psCommand, cancellationToken);
                 }
 
+                // If we're in a breakpoint it means we're executing either interactive commands in
+                // a debug prompt, or our own special commands to query the PowerShell debugger for
+                // state that we sync with the LSP debugger. The former commands we want to send
+                // through PowerShell's `Debugger.ProcessCommand` so that they work as expected, but
+                // the latter we must not send through it else they pollute the history as this
+                // PowerShell API does not let us exclude them from it.
                 return _pwsh.Runspace.Debugger.InBreakpoint
-                    && (IsDebuggerCommand(_psCommand) || _pwsh.Runspace.RunspaceIsRemote)
+                    && (PowerShellExecutionOptions.AddToHistory || IsPromptCommand(_psCommand))
```

Comment:
```suggestion
                    && (PowerShellExecutionOptions.AddToHistory || IsPromptCommand(_psCommand) || _pwsh.Runspace.RunspaceIsRemote)
```

All remote invocations still need to go through the debugger when it's stopped since we can't be on the out of process thread.